### ISSUE TITLE
leptos_config::get_configuration doesn't support workspace metadata in Cargo.toml for multiple projects

### ISSUE THREAD

**Issue by brofrain** [leptos-rs/leptos#2700]
**Describe the bug**
Using `Cargo.toml` to extract Leptos configuration at runtime doesn't work for workspaces with multiple projects defined. `leptos_config::get_configuration` returns an error instead due to an incorrect TOML format.

**Leptos Dependencies**

```toml
leptos = { version = "0.6" }
leptos_meta = { version = "0.6" }
leptos_router = { version = "0.6" }
leptos_axum = { version = "0.6" }
```

The problem applies to 0.7 versions as well.

**To Reproduce**
1. Setup the [workspace](https://github.com/leptos-rs/start-axum-workspace) starter:
    ```bash
     cargo leptos new --git https://github.com/leptos-rs/start-axum-workspace/
    ```
2. Copy-paste the `[[workspace.metadata.leptos]]` section in `Cargo.toml`. Change the `name` of the new Leptos project e.g. "start-axum-workspace-copy"
3. In `server/src/main.rs` change this line `let conf = get_configuration(None).await.unwrap();` to `let conf = get_configuration(Some("Cargo.toml")).await.unwrap();`
4. Run either of those projects:
    ```bash
    cargo leptos watch --project start-axum-workspace
    ```
    or:
    ```
    cargo leptos watch --project start-axum-workspace-copy
    ```
5. The application will panic at runtime with something like this:
    ```
    thread 'main' panicked at server/src/main.rs:18:60:
    called `Result::unwrap()` on an `Err` value: ConfigError("TOML parse error at line 103, column 1\n    |\n103 | name = \"start-axum-workspace-copy\"\n    | ^\nduplicate key `name` in document root\n")
    ```

**Expected behavior**
`leptos_config::get_configuration` should properly extract the configuration of the project being currently run. It may potentially need an additional argument specifying the project's name to be passed.

**Additional context**
`leptos_config::get_config_from_str` doesn't implement any kind of multi-project support. It just flattens Leptos configuration as root-level properties before deserializing the TOML string. This causes a parsing error due to multiple duplicate fields being defined.

I can submit a PR fixing that, but it may take a week or a few before I have the necessary time.

**Comment by benwis** [comment]
So the intention with get_configuration is that it's meant to indicate whether you're using cargo_leptos or not. So you'd normally pass None if you're doing a workspace project, and then it detects the correct values from the tool.

**Comment by brofrain** [comment]
Hi @benwis,

I believe this behavior could be improved even if it is indeed correct. I might have missed something, but I don't recall passing `None` in `get_configuration` being documented anywhere as a requirement for workspace projects. My main concern is how I got this error and that it doesn't panic gracefully with "you're doing something wrong", but feels more like a bug instead.

My use case is a Shuttle project based on https://github.com/leptos-community/deployment-ssr-shuttle, which is referenced in the [book](https://book.leptos.dev/deployment/ssr.html?highlight=shuttle#deploy-to-shuttlers). The only difference really is that I use it in a workspace along with a few other apps. My Shuttle project broke literally the moment I added a new Leptos app to the workspace. This feels a bit inconsistent and counterintuitive. If it is an anti-pattern on my side, then I would expect it to break much sooner.

Shuttle doesn't allow env variables and enforces using its secrets instead. Using `Cargo.toml` for Leptos configuration seems like a very convenient alternative, which is why I'm trying to implement this approach.

It would be nice in my opinion to tweak it a little in one of the following ways:
- preserve the current behavior, but panic with a more readable message instead. I personally don't like it, because it doesn't feel consistent that a single workspace project just works, and adding more breaks it. It is still better than the current "TOML parse error" though.
- removing workspace support completely from `get_configuration` and handling workspace metadata with a message suggesting using env vars instead.
- adding something like `get_configuration_for_project(<path>, <project name>)` that would correctly parse workspace Leptos metadata defined for multiple projects.

I can work on implementing that. Just need some guidance on which direction would be most preferred 🙂
